packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -327,4 +343,50 @@ App.propTypes = {
     config: PropTypes.object
 }
 
+/**
+ * a hook that parallelly and individually fetches category based on the given ids
+ * @param ids - list of categories ids to fetch
+ * @param queryOptions -  react query options
+ * @return list of react query results
+ */
+export const useCategoryBulk = (ids, queryOptions) => {
+    if (!ids || ids.length === 0) {
+        return
+    }
+    const api = useCommerceApi()
+    const {getTokenWhenReady} = useAccessToken()
+    const {
+        app: {commerceAPI}
+    } = getConfig()
+    const {
+        parameters: {organizationId}
+    } = commerceAPI
+    const {site} = useMultiSite()
+
+    const queries = ids.map((id) => {
+        return {
+            queryKey: [
+                '/organizations/',
+                organizationId,
+                '/categories/',
+                id,
+                {id, levels: 2, organizationId, siteId: site.id}
+            ],
+            queryFn: async () => {
+                const token = await getTokenWhenReady()
+                const res = await api.shopperProducts.getCategory({
+                    parameters: {id, levels: 2},
+                    headers: {
+                        authorization: `Bearer ${token}`
+                    }
+                })
+                return res
+            },
+            ...queryOptions
+        }
+    })
+    const res = useQueries({queries})
+    return res
```

Comment:
```suggestion
    return useQueries({queries})
```
🤷

Reply:
I like to assign a variable so if i want to see the res, I just need to console it out :P